src/components/Components/ComponentDetails/tabs/ComponentRegistryLogin.tsx

```diff
@@ -1,11 +1,24 @@
 import * as React from 'react';
-import { Alert, Skeleton } from '@patternfly/react-core';
+import { Alert, ClipboardCopy, Skeleton, Text, TextContent } from '@patternfly/react-core';
+import { useAuth } from '~/auth/useAuth';
 import ExternalLink from '~/shared/components/links/ExternalLink';
 import { getErrorState } from '~/shared/utils/error-utils';
 import { useRegistryLoginUrl } from './useRegistryLoginUrl';
 
+/**
+ * Extracts the registry URL from the OAuth URL by removing the protocol and /oauth path
+ * @param registryLoginUrl - The full OAuth URL (e.g., "https://registry.example.com/oauth")
+ * @returns The registry URL without protocol and OAuth path (e.g., "registry.example.com")
+ */
+const getRegistryUrlFromOAuthUrl = (registryLoginUrl: string): string => {
+  return registryLoginUrl.replace(/^https?:\/\//, '').replace(/\/oauth\/?$/, '');
```

Comment:
what you are looking for here is the hostname. No need to use RegEx here use `URL` class
```javascript
const url = new URL("https://google.com/");
url.hostname // google.com
```